src/lib/utils/memory_resource_manager.cpp

```diff
@@ -0,0 +1,30 @@
+#include "utils/memory_resource_manager.hpp"
+
+namespace opossum {
+
+const tbb::concurrent_vector<ResourceRecord>& MemoryResourceManager::memory_resources() const {
+  return _memory_resources;
+}
+
+boost::container::pmr::memory_resource* MemoryResourceManager::get_memory_resource(
+    const OperatorType operator_type, const std::string& operator_data_structure) {
+  // Return default memory resource if tracking is disabled.
+  if (!_tracking_is_enabled) {
+    return boost::container::pmr::get_default_resource();
+  }
+
+  auto resource_ptr = new TrackingMemoryResource(); // NOLINT
```

Comment:
Could this lead to a memory leak? How is it freed?

Reply:
Indeed it does.
The memory leaks will be fixed when we implement a cleanup strategy.

Reply:
Alright, true 👍

Reply:
If you need a raw pointer here, you can still achieve automatic memory management by creating the object and wrapping it in a smart pointer. Subsequently, you can get the raw pointer via the smart pointers `get` function.

Reply:
Thanks for the comment. We will make use of this when implementing the cleanup :)